proteus/Gauges.py

```diff
@@ -7,14 +9,84 @@
 from . import Comm
 from .AuxiliaryVariables import AV_base
 from .Profiling import logEvent as log
+from . import Geom
 
-class PointGauges(AV_base):
+def PointGauges(gauges, activeTime=None, sampleRate=0, fileName='point_gauges.csv'):
+    """ Create a set of point gauges that will automatically be serialized as CSV data to the requested file.
+
+    :param gauges: An iterable of "gauges".  Each gauge is specified by a 2-tuple, with the first element in the
+    tuple a set of fields to be monitored, and the second element a tuple of the 3-space representations of the gauge
+    locations.
+
+    See the Gauges class for an explanation of the other parameters.
+
+    Example:
+
+    p = PointGauges(gauges=((('u', 'v'), ((0.5, 0.5, 0), (1, 0.5, 0))),
+                            (('p',), ((0.5, 0.5, 0),))),
+                    activeTime=(0, 2.5),
+                    sampleRate=0.2,
+                    fileName='combined_gauge_0_0.5_sample_all.csv')
+
+    This creates a PointGauges object that will monitor the u and v fields at the locations [0.5, 0.5,
+    0] and [1, 0.5, 0], and the p field at [0.5, 0.5, 0] at simulation time between = 0 and 2.5 with samples
+    taken no more frequently than every 0.2 seconds.  Results will be saved to:
+    combined_gauge_0_0.5_sample_all.csv.
+    """
+
+    # build up dictionary of location information from gauges
+    # dictionary of dictionaries, outer dictionary is keyed by location (3-tuple)
+    # inner dictionaries contain monitored fields, and closest node
+    # closest_node is None if this process does not own the node
+    points = OrderedDict()
+    fields = list()
+
+    for gauge in gauges:
+        gauge_fields, gauge_points = gauge
+        for field in gauge_fields:
+            if field not in fields:
+                fields.append(field)
+        for point in gauge_points:
+            # initialize new dictionary of information at this location
+            if point not in points:
+                l_d = {'fields': set()}
+                points[point] = l_d
+            # add any currently unmonitored fields
+            points[point]['fields'].update(gauge_fields)
+    return Gauges(fields, activeTime, sampleRate, fileName, points=points)
+
+
+def LineGauges(gauges, activeTime=None, sampleRate=0, fileName='line_gauges.csv'):
+    """ Create a set of line gauges that will automatically be serialized as CSV data to the requested file.
```

Comment:
If I understand the code below, all line gauges are now line integral gauges. We're going to need both line integral gauges and line gauges that store the profile of the field over a line.